x/finality/keeper/msg_server.go

```diff
@@ -303,3 +309,39 @@ func (k Keeper) slashFinalityProvider(ctx context.Context, fpBtcPk *bbn.BIP340Pu
 		panic(fmt.Errorf("failed to emit EventSlashedFinalityProvider event: %w", err))
 	}
 }
+
+// validateActivationHeightAddFinalitySig returns error if the msg add finality
+// block height is lower than the activation height
+func (ms msgServer) validateActivationHeightAddFinalitySig(ctx sdk.Context, msg *types.MsgAddFinalitySig) error {
+	// TODO: remove it after Phase-2 launch in a future coordinated upgrade
+	activationHeight := ms.GetActivationHeight(ctx)
+	if msg.BlockHeight < activationHeight {
+		ms.Logger(ctx).With(
+			"finalityBlockHeight", msg.BlockHeight,
+			"activationHeight", activationHeight,
+		).Info("BTC finality is not activated yet")
+		return types.ErrFinalityNotActivated.Wrapf(
+			"finality block height: %d is lower than activation height %d",
+			msg.BlockHeight, activationHeight,
+		)
+	}
+	return nil
+}
+
+// validateActivationHeightCommitPubRand returns error if the msg commit pub rand list
+// start height is lower than the activation height
+func (ms msgServer) validateActivationHeightCommitPubRand(ctx sdk.Context, msg *types.MsgCommitPubRandList) error {
```

Comment:
If we disable pub rand commit before the finality activation height, wouldn't it be possible that upon the finality actviation height, none of the FPs will have timestamped pub rand, such that no FP will be voting. Only after ~10 hours some FPs start having timestamped pub rand and will start voting.

Wdyt about allowing FPs to commit pub rand anytime, and just disabling finality signature until fiunality activation height?

Reply:
I think the point here is that fps won't waste any pub rand as the first one is not lower than the activation height. Fps can adjust the logic for the first commit starting from higher than the activation height but still can submit the commit early

Reply:
finality providers can commit the pub rand for the `start height = finality activation height` at any time and have it timestamped 

> Wdyt about allowing FPs to commit pub rand anytime, and just disabling finality signature until fiunality activation height?

they would commit some pub rand that is not going to be used, not a big problem to be honest 
What do you all think? @gitferry @KonradStaniec

Reply:
just saw @gitferry opinion https://github.com/babylonlabs-io/babylon/pull/204#discussion_r1808533906

Reply:
tbh I would keep it as it is i.e:
- fp can commit randomness at any time (i.e send transaction with commitment any time)
- fp can commit randomness only starting from activation height (i.e lowest commit must have at least start height >= activation height)
so this is what also @gitferry mentioned

Reply:
I see, the logic here is that the start height should be at least activation height, rather than that FP can commit pub rand only after the activation height. That makes sense :+1: sorry for the confusion